### ISSUE TITLE
tween interpolation

### ISSUE THREAD

**Issue by blurymind** [dalboris/vpaint#41]
Right now vpaint can only create a linear interpolation. I suggest we have the ability to add a way to control the tween spacing. Here is how that looks like in both tween maker and cacani:
![tweenspacing](https://cloud.githubusercontent.com/assets/6495061/10150753/10ea8646-664a-11e5-93b5-6e152fa63297.PNG)

When the time line cursor is at a tween, the user can adjust the tween's spacing and add a slow up at the start, the end, or both. This is especially important for animation.

**Comment by dalboris** [comment]
Yes, thx for this suggestion! Control over the timing of the motion is definitelty important. Also, control over the trajectory taken by tweens might be added. In 3D animation, this is often done by tweaking the tangents of animation curves, which gives you even more precise control than this "tween spacing" tool. I wonder how well could this concept be ported to 2D digital animation.

**Comment by blurymind** [comment]
You could do it with bezier curves to represent the spacing. You are right that this approach is not as flexible. I liked it because it takes much less steps to control and it represents spacing in a clearer way than curves.

I suppose that the best solution for me personally is to have the tween spacing slider control a bezier curve that represents the spacing in the space of the timeline between the two keyframes. But then in order to see that curve you would need much more vertical space for each layer of animated keyframes.

 Perhaps it could show the curve between two keyframes only when you hover your mouse over the the tweened space between the two keyframes (timeline)?
And let you edit the spacing with a tween spacing slider if you click on the tween between the two keyframes (timeline).

**Comment by blurymind** [comment]
An even better better way to do this in the timeline - instead of a bezier curve, use the same straight line like the one in tween spacing editor - only horizontal.
![tweenspacing2](https://cloud.githubusercontent.com/assets/6495061/10188987/4b2d9fb4-676c-11e5-85e5-64e220ee8fc1.png)
This will preserve precious vertical space.

**Comment by dalboris** [comment]
Yes, I think having this control/information in the timeline is where it makes more sense, and having the traditional "tween spacing" should be an option, since trad animators are used to this. Though, in any case, it's not trivial to have a good visualization of it since different edges may have different timing. It is also challenging technically since those edges, with different timing, can be connected to one another. But I'll definitely think about it and try to come with a good solution, all these are interesting and important challenges to solve.

**Comment by blurymind** [comment]
It would be great to have it! :)
For future reference, here is how it looks like in spine and spriter
![features-graph](https://cloud.githubusercontent.com/assets/6495061/10208883/397bb40e-67df-11e5-82f7-4100d1f6f036.png)
http://esotericsoftware.com/spine-in-depth
This is also what most 3d software uses.

I find this way of representing spacing less intuitive and further away from the traditional way most animators have been reading it from the x-sheet.
![animation spacing charts](https://cloud.githubusercontent.com/assets/6495061/10208932/a5c4c2e0-67df-11e5-907f-f3fdb2dd92d3.png)

**Comment by dalboris** [comment]
Yep, no worries, I am very familiar with both representations.
